commit 2bea399ea964f257fe1518229875da0a0f99c33d (refs/users/09/1000009/edit-1307/2, refs/changes/07/1307/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-01-20T15:28:27-08:00 (2 years, 9 months ago)
    
    CMON-36 Add 'cmon-agent' and 'cmon' targets to mountain-gorilla
    Reviewed by: Richard Kiene <richard.kiene@joyent.com>
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Richard Kiene <richard.kiene@joyent.com>

diff --git a/Makefile b/Makefile
index c2a37a6..01d9e5c 100644
--- a/Makefile
+++ b/Makefile
@@ -879,6 +879,72 @@ clean_workflow:
 	(cd build/sdc-workflow && gmake clean)
 
 
+#---- CMON
+
+_cmon_stamp=$(TRITON_CMON_BRANCH)-$(TIMESTAMP)-g$(TRITON_CMON_SHA)
+CMON_BITS=$(BITS_DIR)/cmon/cmon-pkg-$(_cmon_stamp).tar.bz2
+CMON_IMAGE_BIT=$(BITS_DIR)/cmon/cmon-zfs-$(_cmon_stamp).zfs.gz
+CMON_MANIFEST_BIT=$(BITS_DIR)/cmon/cmon-zfs-$(_cmon_stamp).imgmanifest
+
+.PHONY: cmon
+cmon: $(CMON_BITS) cmon_image
+
+$(CMON_BITS): build/triton-cmon
+	@echo "# Build cmon: branch $(TRITON_CMON_BRANCH), sha $(TRITON_CMON_SHA), time `date -u +%Y%m%dT%H%M%SZ`"
+	mkdir -p $(BITS_DIR)
+	(cd build/triton-cmon && NPM_CONFIG_CACHE=$(MG_CACHE_DIR)/npm TIMESTAMP=$(TIMESTAMP) BITS_DIR=$(BITS_DIR) gmake release publish)
+	@echo "# Created cmon bits (time `date -u +%Y%m%dT%H%M%SZ`):"
+	@ls -l $(CMON_BITS)
+	@echo ""
+
+.PHONY: cmon_image
+cmon_image: $(CMON_IMAGE_BIT)
+
+$(CMON_IMAGE_BIT): $(CMON_BITS)
+	@echo "# Build cmon_image: branch $(TRITON_CMON_BRANCH), sha $(TRITON_CMON_SHA), time `date -u +%Y%m%dT%H%M%SZ`"
+	./tools/prep_dataset_in_jpc.sh -i "$(CMON_IMAGE_UUID)" -t $(CMON_BITS) \
+		-o "$(CMON_IMAGE_BIT)" -p $(CMON_PKGSRC) -O "$(MG_OUT_PATH)" \
+		-t $(CMON_EXTRA_TARBALLS) -n $(CMON_IMAGE_NAME) \
+		-v $(_cmon_stamp) -d $(CMON_IMAGE_DESCRIPTION)
+	@echo "# Created cmon image (time `date -u +%Y%m%dT%H%M%SZ`):"
+	@ls -l $$(dirname $(CMON_IMAGE_BIT))
+	@echo ""
+
+cmon_publish_image: $(CMON_IMAGE_BIT)
+	@echo "# Publish cmon image to Triton Updates repo."
+	$(UPDATES_IMGADM) import -ddd -m $(CMON_MANIFEST_BIT) -f $(CMON_IMAGE_BIT)
+
+clean_cmon:
+	$(RM) -rf $(BITS_DIR)/cmon
+	(cd build/triton-cmon && gmake clean)
+
+
+#---- cmon agent
+
+_cmon_agent_stamp=$(TRITON_CMON_AGENT_BRANCH)-$(TIMESTAMP)-g$(TRITON_CMON_AGENT_SHA)
+CMON_AGENT_BIT=$(BITS_DIR)/cmon-agent/cmon-agent-$(_cmon_agent_stamp).tgz
+CMON_AGENT_MANIFEST_BIT=$(BITS_DIR)/cmon-agent/cmon-agent-$(_cmon_agent_stamp).manifest
+
+.PHONY: cmon-agent
+cmon-agent: $(CMON_AGENT_BIT)
+
+$(CMON_AGENT_BIT): build/triton-cmon-agent
+	@echo "# Build cmon-agent: branch $(TRITON_CMON_AGENT_BRANCH), sha $(TRITON_CMON_AGENT_SHA), time `date -u +%Y%m%dT%H%M%SZ`"
+	mkdir -p $(BITS_DIR)
+	(cd build/triton-cmon-agent && NPM_CONFIG_CACHE=$(MG_CACHE_DIR)/npm TIMESTAMP=$(TIMESTAMP) BITS_DIR=$(BITS_DIR) gmake release publish)
+	@echo "# Created cmon-agent bits (time `date -u +%Y%m%dT%H%M%SZ`):"
+	@ls -l $(CMON_AGENT_BIT) $(CMON_AGENT_MANIFEST_BIT)
+	@echo ""
+
+cmon-agent_publish_image: $(CMON_AGENT_BIT)
+	@echo "# Publish cmon-agent image to SDC Updates repo."
+	$(UPDATES_IMGADM) import -ddd -m $(CMON_AGENT_MANIFEST_BIT) -f $(CMON_AGENT_BIT)
+
+clean_cmon_agent:
+	$(RM) -rf $(BITS_DIR)/cmon-agent
+	(cd build/sdc-cmon-agent && gmake clean)
+
+
 #---- VMAPI
 
 _vmapi_stamp=$(SDC_VMAPI_BRANCH)-$(TIMESTAMP)-g$(SDC_VMAPI_SHA)
diff --git a/targets.json.in b/targets.json.in
index cc58457..008f05c 100644
--- a/targets.json.in
+++ b/targets.json.in
@@ -558,6 +558,42 @@ cat <<EOF
     ]
   },
 
+  "cmon": {
+    "appliance": "true",
+    "image_name": "cmon",
+    "image_description": "Triton Container Monitor",
+    "image_version": "1.0.0",
+    "// image_uuid": "sdc-minimal-multiarch-lts@15.4.1",
+    "image_uuid": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
+    "pkgsrc": [
+        "coreutils-8.23nb2",
+        "gsed-4.2.2nb4"
+    ],
+    "repos": [
+        {"url": "git@github.com:joyent/triton-cmon.git"}
+    ],
+    "public": true,
+    "deps": [
+      "config-agent",
+      "registrar",
+      "https://download.joyent.com/pub/build/sdcnode"
+    ],
+    "tarballs": [
+      {"tarball": "registrar/registrar-pkg-*.tar.bz2", "sysroot": "/"},
+      {"tarball": "config-agent/config-agent-*.tar.bz2", "sysroot": "/opt/smartdc"}
+    ]
+  },
+
+  "cmon-agent": {
+    "repos": [
+        {"url": "git@github.com:joyent/triton-cmon-agent.git"}
+    ],
+    "public": true,
+    "deps": [
+      "https://download.joyent.com/pub/build/sdcnode"
+    ]
+  },
+
   "vmapi": {
     "appliance": "true",
     "build_platform": "20141030T081701Z",
@@ -1392,6 +1428,7 @@ cat <<EOF
     "deps": [
       "agents_core",
       "hagfish-watcher",
+      "cmon-agent",
       "cn-agent",
       "net-agent",
       "vm-agent",
